packages/teleport/src/Nodes/AddNode/Iam/Iam.tsx

```diff
@@ -0,0 +1,267 @@
+import React from 'react';
+import styled from 'styled-components';
+import { DialogContent, DialogFooter } from 'design/Dialog';
+import {
+  Text,
+  Box,
+  ButtonPrimary,
+  Link,
+  Alert,
+  ButtonLink,
+  ButtonSecondary,
+} from 'design';
+import TextSelectCopy from 'teleport/components/TextSelectCopy';
+import FieldInput from 'shared/components/FieldInput';
+import Validation, { Validator } from 'shared/components/Validation';
+import { Attempt } from 'shared/hooks/useAttemptNext';
+import { Rule } from 'teleport/services/joinToken';
+import DownloadLinks from 'teleport/components/DownloadLinks';
+import { Pencil } from 'design/Icon';
+
+export default function Iam({
+  token,
+  expiry,
+  attempt,
+  onGenerate,
+  isEnterprise,
+  version,
+  onClose,
+}: Props) {
+  const { hostname, port } = window.document.location;
```

Comment:
This should probably use the `react-router` `useLocation` as it covers a few edge cases and allows it to easier be mocked for tests.

Reply:
Hmmm, `location` doesn't seem to have hostname and port, am I missing something?

https://v5.reactrouter.com/web/api/location

Reply:
Hmm sorry I may have steered you wrong...I was sure that those values were in there but I guess not!